pkg/ctl/delete/cluster.go

```diff
@@ -76,75 +100,46 @@ func doDeleteCluster(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg stri
 		return err
 	}
 
-	var deletedResources []string
+	stackManager := ctl.NewStackManager(cfg)
 
-	handleIfError := func(err error, name string) bool {
+	ctl.MaybeDeletePublicSSHKey(meta.Name)
+
+	kubeconfig.MaybeDeleteConfig(meta)
+
+	if hasDeprectatedStacks, err := deleteDeprecatedStacks(stackManager); hasDeprectatedStacks {
 		if err != nil {
-			logger.Debug("continue despite error: %v", err)
-			return true
+			return err
 		}
-		logger.Debug("deleted %q", name)
-		deletedResources = append(deletedResources, name)
-		return false
+		return nil
 	}
 
-	// We can remove all 'DeprecatedDelete*' calls in 0.2.0
-
-	stackManager := ctl.NewStackManager(cfg)
-
 	{
-		tryDeleteAllNodeGroups := func(force bool) error {
-			errs := stackManager.WaitDeleteAllNodeGroups(force)
-			if len(errs) > 0 {
-				logger.Info("%d error(s) occurred while deleting nodegroup(s)", len(errs))
-				for _, err := range errs {
-					logger.Critical("%s\n", err.Error())
-				}
-				return fmt.Errorf("failed to delete nodegroup(s)")
-			}
-			return nil
-		}
-		if err := tryDeleteAllNodeGroups(false); err != nil {
-			logger.Info("will retry deleting nodegroup")
+		tasks, err := stackManager.DeleteTasksForClusterWithNodeGroups(wait, func(_ chan error, _ string) error {
 			logger.Info("trying to cleanup dangling network interfaces")
```

Comment:
this cleanup function is being called once for each nodegroup now. it looks like it was intended to be just run once previously?

Reply:
Yeah, that's true, but I think it'd be okay for now, we need to do more here anyway (see #536). It shouldn't hurt to call this one multiple times.